website/blog/2023-10-01-databend-weekly-113.md

```diff
@@ -0,0 +1,119 @@
+  - name: ZhiHanZ
+  - name: zhyass
+authors:
+  - name: PsiACE
+    url: https://github.com/psiace
+    image_url: https://github.com/psiace.png
+---
+
+[Databend](https://github.com/datafuselabs/databend) is a modern cloud data warehouse, serving your massive-scale analytics needs at low cost and complexity. Open source alternative to Snowflake. Also available in the cloud: <https://app.databend.com> .
+
+## What's On In Databend
+
+Stay connected with the latest news about Databend.
+
+### Loading to Table with Extra Columns
+
+By default, COPY INTO loads data into a table by matching the order of fields in the file to the corresponding columns in the table. It's essential to ensure that the data aligns correctly between the file and the table.
+
+![load extra](https://databend.rs/assets/images/load-extra-0b76668838a386aace5463d65547a40f.png)
+
+If your table has more columns than the file, you can specify the columns into which you want to load data.
+
+When working with CSV format, if your table has more columns than the file and the additional columns are at the end of the table, you can load data using the `FILE_FORMAT` option `ERROR_ON_COLUMN_COUNT_MISMATCH`.
+
+If you are interested in learning more, please check out the resources listed below.
+
+- [Docs | Example 5: Loading to Table with Extra Columns](https://databend.rs/doc/sql-commands/dml/dml-copy-into-table#example-5-loading-to-table-with-extra-columns)
+
+## Code Corner
+
+Discover some fascinating code snippets or projects that showcase our work or learning journey.
+
+### Introducing Read Policies to Parquet Reader
+
+There is a drawback of using `arrow-rs` APIs: when we try to prefetching data for prewhere and topk push downs, we can't reuse the deserialized blocks.
+
+In order to improve the logic of row group reading and reuse the prefetched data blocks at the final output stage, we have done a lot of refactoring and introduced some read policies.
+
+**NoPrefetchPoliy** 
+
+There is no prefetch stage, we can read, deserialize and output data block we need directly.
+
+**PredicateAndTopkPolicy**
+
+At prefetch stage, prefetch columns needed by prewhere and topk. Deserialize them into `DataBlock` and evaluate it into `RowSelection`. Slice the `DataBlock` by batch size and store the `VecDeque` in memory.
+
+At final stage, read remain columns by `RowSelection` and output `DataBlock`s by batch size, and merge the prefetch block and project the block by `output_schema`.
+
+**TopkOnlyPolicy**
+
+It's similar to `PredicateAndTopkPolicy`, but we only evaluate topk at prefech stage.
+
+If you are interested in learning more, please check out the resources listed below:
+
+- [PR #13020 | refactor: introduce ReadPolicy to parquet reader](https://github.com/datafuselabs/databend/pull/13020)
+
+## Highlights
+
+We have also made these improvements to Databend that we hope you will find helpful:
+
+- Added spill info to query log.
+- Added support for compression when using `COPY INTO`.
```

Comment:
Could you pls send me the PR link? Thanks.

Reply:
https://github.com/datafuselabs/databend/pull/13067